File: LeetCode/739. 每日温度(单调栈).py
class Solution:
    def dailyTemperatures(self, T):
        ans = []
        length = len(T)
        for i in range(length):
            ans.append(0)
        stack = []
        for i in range(length):
            while stack and T[stack[-1]] < T[i]:
                # print(stack[-1])
                # ans[stack[-1]] = i - stack.pop()
                small = stack.pop()
                ans[small] = i - small
            stack.append(i)
        return ans 

l = [73,74,75,71,69,72,76,73]
s = Solution()
print(s.dailyTemperatures(l))

File: Week_04/739. 每日温度(单调栈).java
import java.util.Stack;

class Solution {
    public int[] dailyTemperatures(int[] T) {
        int[] ans = new int[T.length];
        Stack<Integer> stack = new Stack<>();

        for (int i = 0;i < T.length;i ++) {
            while (!stack.isEmpty() && T[stack.peek()] < T[i]) {
                ans[stack.peek()] = i - stack.pop();
            }
            stack.push(i);
        }
        return ans;
    }
}